### PR TITLE
Adding rebate rate and fee rate.

### LINKED ISSUE #7614: InteractiveBrokersShortableProvider Provides the Borrowing Cost
#### Expected Behavior
InteractiveBrokersShortableProvider provides the borrowing cost (and rebate fee).
since this data is available. E.g:
```
#SYM|CUR|NAME|CON|ISIN|REBATERATE|FEERATE|AVAILABLE|
AAPL|USD|APPLE INC|265598|XXXXXXX31005|5.0700|0.2500|10000000|
```
The borrowing cost to short AAPL is 0.25% per year (`FEERATE` column).

#### Actual Behavior
Not supported.

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by ashutoshrana171**
#### Description
We added rebate rate and fee rate.

We added rebate rate and fee rate provided by InteractiveBrokersShortableProvider by extending the IBKR Shortable reader to parse the rebate and fee rate

#### Related Issue
Close [7416](https://github.com/QuantConnect/Lean/issues/7614)

#### Motivation and Context
 We added rebate and fee rate supported by Interactive Brokers

#### Requires Documentation Change
No

#### How Has This Been Tested?
We used a sample file from InteractiveBrokersShortableProvider and processed it for testing.
Sample file:
[usa.txt](https://github.com/QuantConnect/Lean.DataSource.InteractiveBrokersShortStock/files/14202487/usa.txt)
New File:
[20240207.csv](https://github.com/QuantConnect/Lean.DataSource.InteractiveBrokersShortStock/files/14202492/20240207.csv)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [X] I have added tests to cover my changes. 
- [X] All new and existing tests passed.
- [X] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Review by AlexCatarino — APPROVED** [review]
Thank you, @ashutoshrana171 !

File: DataProcessing/DataConverter.cs [hunk 1]
```diff
@@ -130,17 +132,31 @@ private void Process(DateTime date, string line)
...
+                    _rebateRateIndex = Array.IndexOf(csv, "REBATERATE");
+                    _feeRateIndex = Array.IndexOf(csv, "FEERATE");
                 }
                 return;
             }
 
             var ticker = csv[_symbolIndex].Replace(' ', '.');
             var borrowableShares = csv[_availableIndex];
+            var rebateRateIndex = csv[_rebateRateIndex];
+            var feeRateIndex = csv[_feeRateIndex];
```
Comment (Martin-Molinero) [inline DataProcessing/DataConverter.cs]: We shouldn't assume `_rebateRateIndex/_feeRateIndex` will be there always, can happen for older dates, so here we should check if they are different than -1 before trying to access them

File: DataProcessing/DataConverter.cs [hunk 2]
```diff
@@ -282,8 +298,12 @@ public void TryAdd(DateTime date, string borrowableShares)
             /// <returns>List of CSV lines, sorted in ascending order by date</returns>
             public IEnumerable<string> ToCsv()
             {
-                return _entries.OrderBy(kvp => kvp.Key)
-                    .Select(kvp => string.Join(",", kvp.Key.ToString("yyyyMMdd"), kvp.Value));
+                return _entries.OrderBy(entry => entry.Item1)
```
Comment (Martin-Molinero) [inline DataProcessing/DataConverter.cs]: I think now that we have more columns, in the output file we should add a header with the columns names too, for both date and symbol files 👍. Lean will already handle (skip) comment lines starting with `#`  see https://github.com/QuantConnect/Lean/blob/master/Common/Data/Shortable/LocalDiskShortableProvider.cs#L101

**Comment by Martin-Molinero** [comment]
Replaced by https://github.com/QuantConnect/Lean.DataSource.InteractiveBrokersShortStock/pull/3

**Pull request closed without merging**